executor/oci/spec_unix.go

```diff
@@ -155,3 +157,41 @@ func cgroupNamespaceSupported() bool {
 	})
 	return supportsCgroupNS
 }
+
+// genereateCDIOptions creates the OCI runtime spec options for injecting CDI
+// devices. Two options are returned: The first ensures that the CDI registry
+// is initialized with refresh disabled, and the second injects the devices
+// into the container.
+func generateCDIOpts(devices []*pb.CDIDevice) ([]oci.SpecOpts, error) {
+	if len(devices) == 0 {
+		return nil, nil
+	}
+	var dd []string
+	for _, d := range devices {
+		if d == nil {
+			continue
+		}
+		dd = append(dd, d.Name)
+	}
+
+	// Inits the CDI registry and disables auto-refresh.
+	withStaticCDIRegistry := func() oci.SpecOpts {
+		return func(ctx context.Context, _ oci.Client, _ *containers.Container, s *oci.Spec) error {
+			registry := cdi.GetRegistry(cdi.WithAutoRefresh(false))
+			if err := registry.Refresh(); err != nil {
+				// We don't consider registry refresh failure a fatal error.
+				// For instance, a dynamically generated invalid CDI Spec file for
+				// any particular vendor shouldn't prevent injection of devices of
+				// different vendors. CDI itself knows better, and it will fail the
+				// injection if necessary.
+				bklog.G(ctx).Warnf("CDI registry refresh failed: %v", err)
+			}
+			return nil
+		}
+	}
```

Comment:
With the v0.7.1 release you could use:
```suggestion
			_ := cdi.Configure(cdi.WithAutoRefresh(false))
			if err := cdi.Refresh(); err != nil {
				// We don't consider registry refresh failure a fatal error.
				// For instance, a dynamically generated invalid CDI Spec file for
				// any particular vendor shouldn't prevent injection of devices of
				// different vendors. CDI itself knows better, and it will fail the
				// injection if necessary.
				bklog.G(ctx).Warnf("CDI registry refresh failed: %v", err)
			}
			return nil
		}
	}
```

Reply:
See https://github.com/containerd/containerd/pull/10041/files

If this is require as external functionality, we could also consider exporting `WithStaticCDIRegistry` from the `oci` package like we do with `WithCDIDevices`.